app/scripts/services/homePagePreference.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .service("HomePagePreference", function(){
+
+    var HOME_PAGE_PREF_KEY = "origin/home-page-pref/";
+
+    var init = function(user) {
+      HOME_PAGE_PREF_KEY += user.metadata.name;
```

Comment:
This could append the user name multiple times if called more than once. I think it's better to add a method to AuthService that gets the current user (without triggering a new login).

```js
var HOME_PAGE_KEY_PREFIX = "openshift/home-page-preference/";
var getHomePagePreference = function() {
  var user = AuthService.getUser(); // add to method
  if (!user) {
    return null;
  }
  var key = HOME_PAGE_KEY_PREFIX + user;
  // ...
};